python/packages/ai/teams/auth/sso/sso_dialog.py

```diff
@@ -74,5 +76,50 @@ async def _step_one(self, context: WaterfallStepContext) -> DialogTurnResult:
     async def _step_two(self, context: WaterfallStepContext) -> DialogTurnResult:
         token_response = context.result
 
-        # TODO: Dedup token exchange responses
+        if token_response and await self._should_dedup(context.context):
+            context.state.temp.duplicate_token_exchange = True
+            return DialogTurnResult(DialogTurnStatus.Waiting)
+
+        context.state.temp.duplicate_token_exchange = False
```

Comment:
Do you know where this field is being used?

Reply:
also curious about this

Reply:
ok reviewed this. 

First, its basic existance is derived from line 134 of the JS [TeamsSsoBotAuthentication.ts](https://github.com/BMS-geodev/teams-ai/blob/adc7bc5c131e8d9c211b3334717510f80aac78af/js/packages/teams-ai/src/authentication/TeamsSsoBotAuthentication.ts#L134) 

it is used in the afterturn concept that the TS explicitly discusses. As we've reviewed elsewhere, the python implementation does not express the afterturn task the same way. 

it is touched last here:

```
        app.turn('afterTurn', async (context, state) => {
            return state.temp.duplicateTokenExchange !== true;
        });
```
So, I'm not sure that we need it and will test today.

Reply:
test passing when it is removed. 

As we've determined we do not need the same afterturn expression (in the python implementation), then this explicit bool flag is probably not required either. 

I will remove it.

@singhk97 @lilyydu